documentation/analytics/recipes/user-retention.md

```diff
@@ -0,0 +1,118 @@
+
+Day 7 Retention
+
+| Date | Day 7 Retention |
+|---------|----------|
+| 2017-05-16T00:00:00.0000000 |	0.98 |
+| 2017-05-17T00:00:00.0000000 |	0.95 |
+| 2017-05-18T00:00:00.0000000 |	0.98 |
+| 2017-05-19T00:00:00.0000000 |	0.58 |
+| 2017-05-20T00:00:00.0000000 |	0.56 |
+| 2017-05-21T00:00:00.0000000 |	0.64 |
+| 2017-05-22T00:00:00.0000000 |	0.6  |
+| 2017-05-23T00:00:00.0000000 |	0.43 |
+| 2017-05-24T00:00:00.0000000 |	0.43 |
+| 2017-05-25T00:00:00.0000000 |	0.51 |
+| 2017-05-26T00:00:00.0000000 |	0.37 |
+
+
+This table shows that
+* 98% of the people who played the game on May 9th 2017 returned to the game again on May 16th 2017. It is possible that some of the 98% have played the game sometime between the May 9th and May 16th, too, but it does not affect the Day 7 retention.
+* Only 58% of the people playing the game on May 12th opened the game for a session again on May 19th 2017.
+* 37% of the gamers on May 19th played the game again on May 26th.
+
+The resulting retention visualisation is as follows:
+![Day 7 Retention](../../images/analytics/retention/Day7Retention.png)
+
+The visualisation shows that the level of retention is decreasing - which is a bad indicator for the game
+
+The same goes for Day 30 Retention and Day 90 Retention.
+
+## Pre-requisites
+
+Technologies:
+* Azure Data Lake Analytics (ADLA) account and an associated Azure Data Lake Store (ADLS) account
+* Generated messages, e.g. start and stop
+    * Event Hub
+    * AD application
+
+Parameters:
+* ``in``: folder location of all events, e.g. /nether/clustering/geo-location/2017/05/16/{*}.csv
+* ``out``: file path where the output file is going to be stored, e.g. /nether/clustering/geo-location/2017/05/16/{*}.csv
+* ``timeframe``: what period of time are you interested in for user retention; it is typically set at 1, 7, 30 and 90.
+
+
+## Recipe Steps
+
+TODO
+
+### 1 Telemetry Needed from the Game
+
+TODO
+Using the Nether REST API or the provided Client SDKs setup the game to send the following messages at these times.
+
+| Message to send                    | At what time                              |
+|------------------------------------|-------------------------------------------|
+| [start](../message-types/start.md)      | Send at specified time units, e.g. every 5 minutes  |
+
+
+
+### 2 Configuration of Message Processor
+
+TODO
+
+
+### 3 Setup Job Queries
+
+(TODO: Describe with a few sentences what purpose the job queries have (if any). Refer to the queries by linking to them in the source code and explain when they should be run.)
+
+| Query that should be run           | By what service                           |
+|------------------------------------|-------------------------------------------|
+| []()             | Using ADLA (Azure Data Lake Analytics)    |
+
+
+```cs
+
+```
+
+#### 3.1 Setup and Schedule U-SQL script
+
+TODO
+
+### 4 Result Details
+
+Format:
```

Comment:
Every time the job is run, it appends one line to a file depending on what dayXretention you want to get. For instance, you will get for day1retention.csv with lines such as 
| 2017-05-16T00:00:00.0000000 |	0.739 |
| 2017-05-17T00:00:00.0000000 |	0.53 |
| 2017-05-18T00:00:00.0000000 |	0.325 |

For day7retention.csv:
| 2017-05-16T00:00:00.0000000 |	0.98 |
| 2017-05-17T00:00:00.0000000 |	0.95 |
| 2017-05-18T00:00:00.0000000 |	0.98 |

You are partially correct: in the below example it says that 84.9% of the players who played on May 15th (not necessarily started playing) played again on May 16th. For day7retention.csv, it means that 98% of players playing on May 9th (7 days before May 16th) also played on May 16th.